odoo/addons/base/models/ir_translation_code.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
```

Comment:
I wouldn't put this code on a AbstractModel. Instead those can be methods in tools, and translations can just be stored on the registry rather than being set in ormcache. It feeld redudant with bootstrap_translations; to refactor and have only one method?